### DISCUSSION TITLE
Reflex app deployment on GCP Compute Engine

### DISCUSSION THREAD

**Discussion by emibarrod** [reflex-dev/reflex#1604] [Show and tell]
First, you need to create a Compute Engine instance. I recommend Ubuntu 22 because it has python 3.10 by default and saves us a lot of problems. If you don't know how to create it, [read this](https://cloud.google.com/compute/docs/instances/create-start-instance).

Once you create it, make sure that ports 8000 and 3000 are allowed in the firewall. By default, they are not, so we have to create a firewall rules. My rule looks like this:

![image](https://github.com/reflex-dev/reflex/assets/56123279/c9df1a65-ff18-401b-b36f-64643020aa29)

Now, I will assume you used Ubuntu 22 as the OS of the instance.

You have to use ssh to connect to the instance. To do that there are a lot of methods but I'll keep it simple and I will use GCP ssh:

![Paste](https://github.com/reflex-dev/reflex/assets/56123279/7ca1faa7-0837-437c-8e08-9c07b2d40f7a)

Once we are connected, we execute the following commands:

1. `sudo apt-get update`
2. `sudo apt-get install python3-pip nodejs zip`
3. `python3 -m pip install reflex`
4. `PATH=$PATH:/home/<your_username>/.local/bin`

Now clone your repo where your Reflex app is:

6. `git clone https://github.com/<your_repo>`

After cloning the app, we go into the folder and init the app:

8. `cd your_reflex_app/`
9. `reflex init`

Make sure that your app config looks something like this:

```python
import reflex as rx

config = rx.Config(
    app_name="your_app_name",
    api_url="http://<your_external_ip>:8000"
)
```
where external_ip is the external ip of your Compute Engine instance.

To finish, run the app in production environment (you can use just reflex run to develop):

11. `reflex run --env prod`

If something is unclear or missing, please tell me and I will update it.

I hope it helps!

Credits to @picklelo for his [Digital Ocean guide](https://github.com/orgs/reflex-dev/discussions/802). It was a great start to be able to deploy on GCP.

**Comment by Alek99** [comment]
Nice thanks for the tutorial!

**Comment by JeremyLG** [comment]
If you don't want to fill your external_ip manually, you can even find it through python subprocess cmd : 
```python
import subprocess
ip_address = subprocess.Popen(
    [
        'gcloud compute instances list --format json | jq -r ".[].networkInterfaces[0].accessConfigs[0].natIP"'
    ]
)
```

**Comment by psiKb** [comment]
Hi, 

Thank you for the tutorial. I followed the provided instruction and i am able to run my app on GCE. 

However, I am unable to access the reflex-app on my computer. 
On GCE terminal, i have a message
App running at: http://localhost:3000

However on my computer on firefox:

The connection was reset

The connection to the server was reset while the page was loading.


Please let me know if i missed some something. 

Thank you

**Comment by psiKb** [comment]
Thanks, I managed to make it work. 

I needed to click on
**Protocols and ports**
allow all ports
